allennlp/notebooks/vocabulary.ipynb

```diff
@@ -0,0 +1,293 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "\n",
+    "Above, we demonstrated the basic functionality of the namespaces in the Vocabulary. So far so good - probably not much different to other `Vocabulary` type classes for NLP that you've seen before. However, we'd ideally like to \n",
+    "generate a full `Vocabulary` without having to individually add all the different words. Below, we'll generate a `Dataset` consisting of a single `Instance` and use it to automatically generate a `Vocabulary`. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from allennlp.data.fields import TextField, TagField\n",
+    "from allennlp.data import Dataset, Instance\n",
+    "from allennlp.data.token_indexers import SingleIdTokenIndexer\n",
+    "sentence = TextField(tokens=[\"Paul\", \"Allen\", \"is\", \"a\", \"great\", \"guy\", \".\"],\n",
+    "                     token_indexers={\"tokens\": SingleIdTokenIndexer()})\n",
+    "tags = TagField([\"PERSON\", \"PERSON\", \"O\", \"O\", \"O\", \"O\", \"O\"], sentence, tag_namespace=\"tags\")\n",
+    "toy_dataset = Dataset([Instance({\"sentence\": sentence, \"tags\": tags})])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "\n",
+    "Now we've generated this baby dataset with one training instance, we can generate a `Vocabulary` using a classmethod on `Vocabulary`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 1/1 [00:00<00:00, 6288.31it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{0: '@@PADDING@@', 1: '@@UNKOWN@@', 2: 'guy', 3: 'is', 4: 'Allen', 5: 'a', 6: 'Paul', 7: 'great', 8: '.'}\n",
+      "{0: 'O', 1: 'PERSON'}\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "vocab = Vocabulary.from_dataset(toy_dataset)\n",
+    "print(vocab.get_index_to_token_vocabulary(\"tokens\"))\n",
+    "print(vocab.get_index_to_token_vocabulary(\"tags\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the vocab we created has `tokens` and `tags` namespaces. These come from the key in the `token_indexers` dict in the `TextField` and the `tag_namespace` parameter in the `TagField`. At first, it seems confusing as to why it's possible to have multiple `TokenIndxers`. This is because in `allennlp`, we make a distinction between _tokenisation_ and _token representation_. More on this in the NLP API Tutorial!"
```

Comment:
Instead of "At first, it seems confusing", I'd say something like "At first, it might seem confusing", or "You might be wondering".  Also, I'd expect this to be explained first in the Field tutorial, not in the NLP API tutorial (probably mentioned again in the NLP API tutorial).